eng/build.sh

```diff
@@ -38,6 +38,10 @@ usage()
   echo "                                  [Default: Debug]"
   echo "  -runtimeFlavor (-rf)            Runtime flavor: CoreCLR or Mono."
   echo "                                  [Default: CoreCLR]"
+  echo "  -primaryRuntime                 Primary Runtime: CoreCLR or Mono. Unlike -runtimeFlavor, this produces a .NET-like"
+  echo "                                  runtime (including tools, apphosts, etc) but uses either Mono or CoreCLR as the"
+  echo "                                  actual runtime. Useful for building .NET on architectures without CoreCLR support,"
+  echo "                                  or emulating the build for those architectures on x64."
```

Comment:
fwiw, I still don't like this name. I had rather had an argument that captures _this produces a .NET runtime_.

We  now have:

To build a .NET-like runtime that uses CoreCLR on x64:
```
./build.sh
```
To build mono on x64:
```
./build.sh -runtimeFlavor Mono
```
To build a .NET-like runtime that uses mono on x64:
```
./build.sh -runtimeFlavor Mono -primaryRuntime Mono
```
Undefined behavior:
```
./build.sh -primaryRuntime Mono
```

Reply:
I agree. Do we really need a separate argument or could we say if you explicitly specify a runtimeFlavor then make that the primary runtime?

Reply:
looks like this PR stalled a bit, what's our current thinking here?

Reply:
I am not sure what the way forward here is.

As far as a I can tell, `./build.sh -primaryRuntime Mono` is the same as `./build.sh -runtimeFlavor Mono`: you get mono and nothing else.

I see @tmds's point about making things more complex, but I am not sure what the desired way forward is. As far as I can

I think making runtimeFlavor the primary runtime is the opposite of what we want. We want to be able to produce a .NET runtime (what this PR was calling primaryRuntime) where the underlying runtime (aka runtimeFlavor) is mono, not CoreCLR.

Reply:
Suggestion: add a flag named `--mono-runtime` with description `Use the Mono runtime instead of the CoreCLR runtime.`.

The build scrip can use this to set `PrimaryRuntimeFlavor`/`RuntimeFlavor` as needed.

For the build script user, I think provides a user-friendly option.

Reply:
@akoeplinger what do you think about the suggestion?

Reply:
I like the idea of `--mono-runtime`! I think "runtime" is a little ambiguous, though - I think we are building the .NET runtime with the Mono VM (instead of the Mono runtime with the Mono VM).

Reply:
Sounds good. I think we could shorten it to just `--mono`.

Reply:
Let's use the same name as source-build: `--use-mono-runtime`.

https://github.com/dotnet/dotnet/blob/2fc4462a30e08fe5eccb6dbc271a4297d2bba75f/build.sh#L12